snuba/datasets/transactions_processor.py

```diff
@@ -123,6 +127,21 @@ def process_message(self, message, metadata) -> Optional[ProcessedMessage]:
         if "geo" not in contexts and isinstance(geo, dict):
             contexts["geo"] = geo
 
+        measurements = data.get("measurements")
+        if measurements is not None:
+            try:
+                (
+                    processed["measurements.key"],
+                    processed["measurements.value"],
+                ) = extract_nested(measurements, lambda value: float(value["value"]))
+            except Exception:
+                # Not failing the event in this case just yet, because we are still
```

Comment:
Would it be better to check each value individually, instead of failing the entire measurement for a single bad value?

Reply:
I would hope that relay ensure that the data going into Kafka is correctly typed. Relay could elide invalid measurements and use event errors to track the elision. cc @dashed

Reply:
Relay will normalize the types in https://github.com/getsentry/relay/pull/724

Reply:
@evanh, I would prefer ensuring the entirety of the field is intact before storing it, thus log exception if the logic I would run when relying on relay normalization. This is only during development. Once it is stable we should remove the try catch and rely on relay normalization.

@markstory sure, this is only during development. AS soon as the relay changes are stable in prod we would remove the try catch. A bad event should stop the pipeline, not be sanitized in snuba. Sanitizing in snuba would mean having the event being stored inconsistently between snuba and nodestore.